### PR TITLE
chore: add changesets

### PR DESCRIPTION
**Pull request by neighborhood999**
This PR adding [changesets](https://github.com/atlassian/changesets) for manage repository _(or monorepo)_, it can automation generate **CHANGELOG** and **RELEASE** and publish to npm.

If change type is `patch`, `minor`, `next` or `major`,. etc, just running `yarn changeset`, it will ask you a series of questions what packages you want to release, then what semver bump type, then it will ask for a summary of the entire changeset, generated summary in final.

Yet another [pre](https://github.com/atlassian/changesets/blob/master/packages/cli/README.md#pre) command plz check changesets document.

> Note: This PR not including install [changeset-bot](https://github.com/apps/changeset-bot) and add `NPM_TOKEN` secret.

### CODE REVIEW

**Comment by bluebill1049** [comment]
I will sort out the Tokens :)

**Comment by neighborhood999** [comment]
> I will sort out the Tokens :)

Nice, shoud I change this PR to "Ready for review"?

**Comment by bluebill1049** [comment]
Yes, please keep to try it out in the next release as well.

File: .changeset/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Changesets
```
Comment (pmaier983) [inline .changeset/README.md:1]: I don't think we want to commit this file. May want to add it to .gitignore if its auto-generated every time?

**Review by pmaier983 — COMMENTED** [review]
I like this idea a-lot! I think additional release information could help decrease the amount of question received. It is more work mainly for Bill tho, so I would leave it up to him to decide if its worth it or not.

File: .changeset/README.md (hunk 1 above)
Comment (neighborhood999) [reply .changeset/README.md:1]: It can be removed and it didn't generate for every time.

Comment (bluebill1049) [reply .changeset/README.md:1]: valid points for both :) let's keep it for now and we can sort it out after. Really excited about this, no more manually `npm publish`. 🥳

Comment (bluebill1049) [reply .changeset/README.md:1]: Hey @neighborhood999, can I ask for a favor? would you mind create a PR in this repo as well: https://github.com/react-hook-form/devtools we can do a quick test there.

Comment (neighborhood999) [reply .changeset/README.md:1]: Sure, did you means change this repo to monorepo?

**Review by JeromeDeLeon — DISMISSED** [review]
I do not much about publishing an NPM. Can someone elaborate this with me, how tedious it is than using `npm publish`? Though I must say that I definitely think that this could improve publishing package way more than before but to what extent? I just want to know. 😬😬

File: .changeset/README.md (hunk 1 above)
Comment (bluebill1049) [reply .changeset/README.md:1]: no, I mean just do another PR at devtools with ChagneSeat, so we can do a test over there with a real patch version. Thanks :)

**Comment by bluebill1049** [comment]
> I do not much about publishing an NPM. Can someone elaborate this with me, how tedious it is than using `npm publish`? Though I must say that I definitely think that this could improve publishing package way more than before but to what extent? I just want to know. 😬😬

it's better to leave at the CI level :) then everyone can publish with a merge.

**Review by w3bdesign — DISMISSED** [review]

File: .changeset/README.md (hunk 1 above)
Comment (neighborhood999) [reply .changeset/README.md:1]: OK

Comment (bluebill1049) [reply .changeset/README.md:1]: Thank you! no rush at all :) take you time with it.

Comment (neighborhood999) [reply .changeset/README.md:1]: I created a PR at https://github.com/react-hook-form/devtools/pull/41, please reviewing and check, thank you

Comment (bluebill1049) [reply .changeset/README.md:1]: What? instant lol... thanks a lot @neighborhood999!

File: package.json [hunk 2]
```diff
@@ -32,7 +32,9 @@
     "prepublishOnly": "npm run lint && npm run lint:types && npm test && npm run build",
     "postrelease": "npm publish && git push --follow-tags",
```
Comment (neighborhood999) [inline package.json:33]: This line can be removed if changeset ready.

Comment (bluebill1049) [reply package.json:33]: plz remove. we are ready! changeset is too good 👍

**Review by bluebill1049 — APPROVED** [review]

**Pull request merged**